### DISCUSSION TITLE
How python library is linked?

### DISCUSSION THREAD

**Discussion by zhqrbitee** [wjakob/nanobind#273] [Q&A]
Hi, 
In my project, I need to create a `type_caster` for `int128_t`, so I'm using some private python APIs: `_PyLong_AsByteArray` and `_PyLong_FromByteArray`.  

However, it causes link issue when compiled using `nanobind_add_module`: 
```
Undefined symbols for architecture arm64:
  "__PyLong_AsByteArray", referenced from:
      ...
  "__PyLong_FromByteArray", referenced from:
      ...
```

I guess it is because `nanobind_add_module` doesn't add link to python library? 

I can work around by adding below line after `nanobind_add_module`:
```
    target_link_options(${TARGET_NAME} PRIVATE "-Wl,-undefined,dynamic_lookup")
```
But want to know is there any recommended way to solve this?

**Comment by wjakob** [comment]
You will only have this issue on macOS, and `-Wl,-undefined,dynamic_lookup` is a reasonable workaround. The (very tiny) downside is that the build then won't use the new dynamic fixups feature, causing the resulting binary to be slightly larger. Alternatively, you could create a PR to add these private symbols to `cmake/darwin-ld-cpython.sym` and `cmake.darwin-ld-pypy.sym`. But private symbols are all going away soon as part of Python C API cleanups, so this is not future-proof.

A more portable way is described here: https://docs.python.org/3/c-api/long.html#c.PyLong_FromString: call the Python `from/to_bytes` function.
